### ISSUE TITLE
noxious-server as a library

### ISSUE THREAD

**Issue by cbgbt** [oguzbilgener/noxious#15]
**Is your feature request related to a problem? Please describe.**
I'm currently invoking `noxious-server` via `std::process::Command` in a Rust module in my CI. This mostly works great!

I did have some issues though when debugging timing of the process coming up and successfully binding to a port, then also ensuring it is reliably killed.

**Describe the solution you'd like**
It might be nice to make the `noxious-server` package both a library and a binary, where the binary is a thin wrapper. I'm thinking that in CI, I'd have a choice to something something `tokio::spawn(noxious_server.serve(...))`

**Describe alternatives you've considered**
This isn't strictly required, and it does increase API surface of the package.

**Comment by cbgbt** [comment]
By the way, I'm happy to contribute code in general, just wanted to share some ideas I'd had while working with `noxious`.

**Comment by cbgbt** [comment]
I started skimming through the code and have realized that this may already be possible via `core`.

**Comment by oguzbilgener** [comment]
Yup, `core` has the server. What it doesn't have is the REST API and the store that keeps the proxies. I can definitely see the benefit of exposing the server as a library as well though. 

Thank you for offering to contribute!
